### PR TITLE
Improve GHA situation

### PR DESCRIPTION
**Pull request by Jasagredo**
# Description

- Builds only with `ghc-9.2.5` on windows
- Builds only on non-draft PRs
- Doesn't build Consensus on GHA
- Inexpensive checks are aggregated into one workflow

### CODE REVIEW

**Review by bolt12 — APPROVED** [review]
Looks good! Thank you! :)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,15 +28,13 @@ jobs:
...
-      matrix:
-        ghc: ["8.10.7", "9.2.5"]
-        os: [ubuntu-latest, windows-latest]
+    if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
 
     env:
+
+      ghc: "9.2.5"
       # Modify this value to "invalidate" the cabal cache.
       CABAL_CACHE_VERSION: "2022-12-30"
```
Comment (coot) [inline .github/workflows/build.yml:39]: I think it's better to still use `matrix` even if we just have a single entry for `ghc`.  In the future we might again have a period of time when we will need to support two versions of `ghc`.

File: .github/workflows/cabal.project.local.Windows [hunk 2]
```diff
@@ -49,6 +49,48 @@ package cardano-client
 package cardano-crypto-praos
   flags: -external-libsodium-vrf
 
+package ouroboros-consensus
```
[the remark this answers is not in the document]
Comment (Jasagredo) [reply .github/workflows/cabal.project.local.Windows:52]: Done!

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -29,12 +29,15 @@ jobs:
...
     runs-on: ${{ matrix.os }}
+    if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
 
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.5"]
-        os: [ubuntu-latest, windows-latest]
+        # Currently Hydra/Cicero build with 8.10.7
+        ghc: ["9.2.5"]
```
[the remark this answers is not in the document]
Comment (Jasagredo) [reply .github/workflows/build.yml]: Done!

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+name: Project checks
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'staging'
```
[the remark this answers is not in the document]
Comment (Jasagredo) [reply .github/workflows/checks.yml:7]: Done

File: .github/workflows/build.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jasagredo) [reply .github/workflows/build.yml:38]: Turns out it is not buildable on 9.2.7 https://github.com/input-output-hk/ouroboros-network/actions/runs/4315807581/jobs/7530703089

**Comment by Jasagredo** [comment]
I went into my Windows/MinGW machine and made sure this builds. Merging

**Comment by Jasagredo** [comment]
bors cancel

**Comment by nrdxp** [comment]
Sorry to just pop in here, but I noticed you are using `std` already, so you might be able to benefit from [std-action](https://github.com/divnix/std-action)

**Comment by Jasagredo** [comment]
> Sorry to just pop in here, but I noticed you are using `std` already, so you might be able to benefit from [std-action](https://github.com/divnix/std-action)

Hi Tim, thanks for the suggestion, but currently we don't (or at least I don't think we do!) use Nix in our GHA workflow. Please correct me otherwise, but I don't think we can leverage that action right now.

**Review by coot — APPROVED** [review]

**Comment by Jasagredo** [comment]
bors cancel

**Comment by Jasagredo** [comment]
bors cancel

**Pull request merged**